### ISSUE TITLE
NameError occurs after update

### ISSUE THREAD

**Issue by mythosil** [Shopify/shopify-app-cli#590]
# Issue summary

`NameError` occurs when update of CLI happens. `RbConfig` can't be found.

```
❯ shopify tunnel stop
Updating shopify-cli...done!
x ngrok tunnel not running
/Users/mythosil/.shopify-app-cli/lib/shopify-cli/core/monorail.rb:76:in `monorail_payload': uninitialized constant #<Class:ShopifyCli::Core::Monorail>::RbConfig (NameError)
        from /Users/mythosil/.shopify-app-cli/lib/shopify-cli/core/monorail.rb:40:in `rescue in log'
        from /Users/mythosil/.shopify-app-cli/lib/shopify-cli/core/monorail.rb:24:in `log'
        from /Users/mythosil/.shopify-app-cli/lib/shopify-cli/core/entry_point.rb:18:in `call'
        from /Users/mythosil/.shopify-app-cli/bin/shopify:37:in `block in <main>'
        from /Users/mythosil/.shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:75:in `handle_abort'
        from /Users/mythosil/.shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:21:in `call'
        from /Users/mythosil/.shopify-app-cli/bin/shopify:36:in `<main>'
```

## Steps to reproduce the problem

Comment this line out (or just wait 1 hour) and run `shopify tunnel stop` in Shopify app project.
https://github.com/Shopify/shopify-app-cli/blob/b64e3c362eca01f45858f4227f545c8c200cf4d8/lib/shopify-cli/core/update.rb#L21

## Cause

After updating, CLI `exec` the actual command with an option `--disable-gems` which makes `RbConfig` invisible.
https://github.com/Shopify/shopify-app-cli/blob/b64e3c362eca01f45858f4227f545c8c200cf4d8/bin/shopify#L1

**Comment by mkevinosullivan** [comment]
Hi @mythosil, what is your development platform?  You should be able to run `shopify system` to obtain those details and post them here, although that uses `RbConfig` too, so may not work 😞

**Comment by mythosil** [comment]
@mkevinosullivan Here's the output from `shopify system` .
```
Shopify App CLI
        INSTALL_DIR = /Users/mythosil/.shopify-app-cli

Ruby (via RbConfig)
  /Users/mythosil/.rbenv/versions/2.6.3/bin/ruby
  x86_64-apple-darwin17.7.0 - RbConfig["host"]
  ruby-2.6.0                - RbConfig["RUBY_VERSION_NAME"]

Commands
  ✓ git, /usr/local/bin/git
  ✓ curl, /usr/bin/curl
  ✓ tar, /usr/bin/tar
  ✓ unzip, /usr/bin/unzip
```

**Comment by mkevinosullivan** [comment]
Very weird - I have essentially the same configuration.  I'll try to reproduce ...

**Comment by mythosil** [comment]
I found easier way to reproduce.
```
❯ shopify system
Shopify App CLI
        INSTALL_DIR = /Users/mythosil/.shopify-app-cli

Ruby (via RbConfig)
  /Users/mythosil/.rbenv/versions/2.5.1/bin/ruby
  x86_64-apple-darwin17.7.0 - RbConfig["host"]
  ruby-2.5.0                - RbConfig["RUBY_VERSION_NAME"]

Commands
  ✓ git, /usr/local/bin/git
  ✓ curl, /usr/bin/curl
  ✓ tar, /usr/bin/tar
  ✓ unzip, /usr/bin/unzip

❯ touch -m -A '-010000' ~/.shopify-app-cli/.git/FETCH_HEAD

❯ shopify system
Updating shopify-cli...done!
Shopify App CLI
        INSTALL_DIR = /Users/mythosil/.shopify-app-cli

Ruby (via RbConfig)
This command ran with ID: 13854
Please include this information in any issues/report along with relevant logs
/Users/mythosil/.shopify-app-cli/lib/shopify-cli/core/monorail.rb:76:in `monorail_payload': uninitialized constant #<Class:ShopifyCli::Core::Monorail>::RbConfig (NameError)
        from /Users/mythosil/.shopify-app-cli/lib/shopify-cli/core/monorail.rb:40:in `rescue in log'
        from /Users/mythosil/.shopify-app-cli/lib/shopify-cli/core/monorail.rb:24:in `log'
        from /Users/mythosil/.shopify-app-cli/lib/shopify-cli/core/entry_point.rb:18:in `call'
        from /Users/mythosil/.shopify-app-cli/bin/shopify:37:in `block in <main>'
        from /Users/mythosil/.shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:75:in `handle_abort'
        from /Users/mythosil/.shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:21:in `call'
        from /Users/mythosil/.shopify-app-cli/bin/shopify:36:in `<main>'
```

**Comment by mkevinosullivan** [comment]
Thanks - I tried that and am still unable to reproduce.

If you run `irb` and enter `RbConfig.ruby`, what does it return?

```
% irb
irb(main):001:0> RbConfig.ruby
=> ?????
irb(main):002:0>
```

**Comment by mythosil** [comment]
```
❯ irb
irb(main):001:0> RbConfig.ruby
=> "/Users/mythosil/.rbenv/versions/2.6.3/bin/ruby"
```

This might help too. 
```
❯ ruby -e 'p RbConfig.ruby'
"/Users/mythosil/.rbenv/versions/2.6.3/bin/ruby"

❯ ruby --disable-gems -e 'puts RbConfig.ruby'
Traceback (most recent call last):
-e:1:in `<main>': uninitialized constant RbConfig (NameError)
```

**Comment by mkevinosullivan** [comment]
Thank you.  I can reproduce that!

Does this work for you?
```
% ruby --disable-gems -e "require 'rbconfig'; puts RbConfig.ruby"
```

**Comment by mythosil** [comment]
Yes, it works.
```
❯ ruby --disable-gems -e "require 'rbconfig'; puts RbConfig.ruby"
/Users/mythosil/.rbenv/versions/2.6.3/bin/ruby
```

**Comment by mkevinosullivan** [comment]
I have a PR submitted to fix this.  In the meantime, if you want to get going, add `require 'rbconfig'` near the top of the following files on your system:
```
/Users/mythosil/.shopify-app-cli/lib/shopify-cli/commands/system.rb
/Users/mythosil/.shopify-app-cli/lib/shopify-cli/context.rb
/Users/mythosil/.shopify-app-cli/lib/shopify-cli/core/monorail.rb
```

**Comment by mythosil** [comment]
It's fixed. Thank you!

**Comment by mkevinosullivan** [comment]
Fixed with PR #599
